src/tr2/game/level.c

```diff
@@ -31,66 +31,85 @@
 #include <libtrx/game/objects/traps/movable_block.h>
 #include <libtrx/log.h>
 #include <libtrx/memory.h>
+#include <libtrx/utils.h>
+#include <libtrx/vector.h>
 #include <libtrx/virtual_file.h>
 
+typedef struct {
+    size_t offset;
+    int32_t size;
+} WAV_ENTRY;
+
 static void M_LoadFromFile(const GF_LEVEL *level);
-static void M_InitialiseSoundEffects(void);
-static void M_CompleteSetup(void);
+static void M_InitialiseSoundEffects(const char *file_name);
+static void M_CompleteSetup(const GF_LEVEL *level);
 
-static void M_InitialiseSoundEffects(void)
+static void M_InitialiseSoundEffects(const char *const file_name)
 {
     BENCHMARK benchmark = Benchmark_Start();
-    const char *const file_name = "data\\main.sfx";
     const char *full_path = File_GetFullPath(file_name);
     LOG_DEBUG("Loading samples from %s", full_path);
-    MYFILE *const fp = File_Open(full_path, FILE_OPEN_READ);
+    VFILE *const fp = VFile_CreateFromPath(full_path);
```

Comment:
While I know that VFile has attractive APIs, this is sketchy, as it'll load the entire file into memory at once as VFile doesn't know how to stream internally. It seems to cause a 50% performance downgrade for this function, going from 2 ms to 3+ ms on each level load.